doc/User_Management/User_Management.md

```diff
@@ -0,0 +1,347 @@
+
+**`userd`'s Points of Interaction:**
+1.  **CONFIG_DB:** Single source of truth for user configuration and global security policies.
+2.  **Core User Files:** `/etc/passwd`, `/etc/shadow`, `/etc/group` for basic user identity.
+3.  **PAM Configuration (`/etc/security/faillock.conf`):** To manage global failed login attempt policies per role via `pam_faillock`.
+
+## 6. High-Level Design
+
+### 6.1 `userd` Daemon
+The `userd` daemon's logic will be expanded to manage security configurations idempotently.
+
+**Startup Consistency Check:**
+* **System Reconciliation:** On startup, `userd` will perform a consistency check to ensure that all local users in the Linux system (`/etc/passwd`, `/etc/shadow`, `/etc/group`) match the definitions in CONFIG_DB.
+* **Cleanup of Unmanaged Users:** Any users found in the Linux system that are not defined in CONFIG_DB (except for system users like `root`, `daemon`, etc.) will be automatically removed to maintain consistency.
+* **CONFIG_DB as Source of Truth:** This ensures that CONFIG_DB remains the single source of truth for user management and prevents configuration drift.
+
+**Default Admin User:**
+* **Init Config Integration:** The `admin` user (or the user specified during compilation) is included in `init_cfg.json` with default settings, ensuring it's always present in CONFIG_DB.
+* **Standard Management:** The admin user is managed through the same CONFIG_DB interface as other users, with no special exception handling required.
+* **Default Configuration:** The admin user in `init_cfg.json` will use the username and password hash from the `USERNAME` and `PASSWORD` environment variables during build, with fallback defaults if not specified.
+
+NOTE: If there are no `administrator` role users defined in `CONFIG_DB`, `userd` will create an `admin` user with default settings if the `admin` user does not exist, otherwise will ensure the user account is enabled.
+
+**Configuration File Hierarchy:**
+* **`init_cfg.json`:** Contains base system defaults including the default admin user. This file is generated at build-time from a Jinja2 template and is always present. It ensures every SONiC system has essential user management configuration.
+* **`golden_config_db.json`:** Optional site-specific configuration file that can override defaults when used with `config load_minigraph --override_config`. This allows deployments to customize user configurations without modifying the base system defaults.
+* **Configuration Precedence:** When both files are present, `golden_config_db.json` takes precedence over `init_cfg.json` during minigraph reload operations with override enabled.
+
+**User Account Management:**
+* **Account Status Control:** `userd` will monitor the `enabled` attribute for each user. When `enabled` is `false`, it will prepend `!` to the password hash in `/etc/shadow` to disable password-based login. When `enabled` is `true`, it will restore the original password hash.
```

Comment:
What is the goal of this functionality?
If we want to disable password-based login, we can have different knobs in ssh-server and other entry points rather than altering the /etc/shadow file.